### PR TITLE
[Feature][57986] Create Tar Bar

### PR DESCRIPTION
**Pull request by DoQuangKhanh0209**
## Related Tickets
- [https://edu-redmine.sun-asterisk.vn/issues/57987](url)
- [https://edu-redmine.sun-asterisk.vn/issues/57988](url)
## WHAT
- Create UI
- Handle Event 
## Evidence (Screenshot or Video)

https://user-images.githubusercontent.com/114377716/208055595-1769a37f-bbad-4969-a2a3-1cbba0270b6f.MP4

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        } else if tag == 3 {
+            guard let searchView = self.storyboard?.instantiateViewController(withIdentifier: "SearchViewController") as? SearchViewController else { return }
+            contentView.addSubview(searchView.view)
+            searchView.didMove(toParent: self)
+        } else {
+            guard let favouriteView = self.storyboard?.instantiateViewController(withIdentifier: "FavouriteViewController") as? FavouriteViewController else { return }
+            contentView.addSubview(favouriteView.view)
+            favouriteView.didMove(toParent: self)
+        }
+     }
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift:57]: swift case

File: Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        designableTabBar()
+        Timer.scheduledTimer(withTimeInterval: 0.1, repeats: false) { (time) in
+            self.forHome()
+        }
+    }
+    
+    private func forHome() {
```
Comment (namtn-2711) [inline Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift]:
```suggestion
    private func toHome() {
```

File: Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+//
+//  Created by Quang Khanh on 14/12/2022.
+//
+
+import UIKit
+
+final class TarbarViewController: UIViewController {
+
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var viewforTab: UIView!
```
Comment (namtn-2711) [inline Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift]:
```suggestion
    @IBOutlet private weak var tabbarView: UIView!
```

File: Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    
+    private func designableTabBar() {
+        viewforTab.layer.cornerRadius = viewforTab.frame.size.height / 2
+        viewforTab.clipsToBounds = true
+    }
+    
+    @IBAction private func oneClickTabBar(_ sender: UIButton) {
+        let tag = sender.tag
+        switch tag {
```
Comment (namtn-2711) [inline Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift]:
```suggestion
        switch sender.tag {
```

File: Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+final class TarbarViewController: UIViewController {
+
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var viewforTab: UIView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        designableTabBar()
+        Timer.scheduledTimer(withTimeInterval: 0.1, repeats: false) { (time) in
```
Comment (ghost) [inline Making-Cocktails/Making-Cocktails/Controllers/TarbarViewController.swift]: retain cycle

**Pull request merged**